src/modules/endpoint/sagas/__tests__/loadCityContent.spec.js

```diff
@@ -0,0 +1,279 @@
+  const eventsBuilder = new EventModelBuilder('loadCityContent-events', 2)
+
+  const categories = categoriesBuilder.build()
+  const cities = new CityModelBuilder(1).build()
+  const languages = new LanguageModelBuilder(2).build()
+  const events = eventsBuilder.build()
+  const resources = { ...categoriesBuilder.buildResources(), ...eventsBuilder.buildResources() }
+  const fetchMap = { ...categoriesBuilder.buildFetchMap(), ...eventsBuilder.buildFetchMap() }
+
+  await dataContainer.setEvents(city, language, events)
+  await dataContainer.setCategoriesMap(city, language, categories)
+  await dataContainer.setLanguages(city, languages)
+  await dataContainer.setResourceCache(city, language, resources)
+  await dataContainer.setCities(cities)
+
+  return { languages, cities, categories, events, resources, fetchMap }
+}
+
+jest.mock('moment-timezone', () => {
+  const moment = jest.requireActual('moment-timezone')
+  const mockActualTz = moment.tz
+  moment.tz = jest.fn(mockActualTz)
+  return moment
+})
+
+const mockTz = (mockDate: moment) => {
+  const previous = moment.tz.getMockImplementation()
+  moment.tz.mockImplementation(function (): moment {
+    if (arguments.length <= 1) {
+      return mockDate
+    }
+
+    return previous(...arguments)
+  })
+
+  return { restore: () => { moment.tz.mockImplementation(previous) } }
+}
+
+describe('loadCityContent', () => {
+  const mockDate = jest.requireActual('moment')('2000-01-05T00:11:00.000Z')
+  const lastUpdate = moment('2000-01-05T10:10:00.000Z')
+
+  let restoreMoment
+
+  beforeEach(async () => {
+    RNFetchBlob.fs._reset()
+    await AsyncStorage.clear()
+
+    const { restore } = mockTz(mockDate)
+    restoreMoment = restore
+  })
+
+  afterEach(async () => {
+    restoreMoment()
+  })
+
+  const city = 'augsburg'
+  const language = 'en'
+
+  it('should set selected city when not peeking', async () => {
+    const dataContainer = new DefaultDataContainer()
+    await prepareDataContainer(dataContainer, city, language)
+
+    await dataContainer.setLastUpdate(city, language, lastUpdate)
+
+    await expectSaga(loadCityContent,
+      dataContainer, city, language, new ContentLoadCriterion({
+        forceUpdate: false,
+        shouldRefreshResources: true
+      }, false)
+    ).run()
+
+    expect(await new AppSettings().loadSelectedCity()).toBe('augsburg')
+    expect(await dataContainer.getLastUpdate(city, language)).toBe(lastUpdate)
+  })
+
+  it('should not set selected city when peeking', async () => {
+    const dataContainer = new DefaultDataContainer()
+    await prepareDataContainer(dataContainer, city, language)
+
+    await dataContainer.setLastUpdate(city, language, lastUpdate)
+
+    await expectSaga(loadCityContent,
+      dataContainer, city, language, new ContentLoadCriterion({
+        forceUpdate: false,
+        shouldRefreshResources: true
+      }, true)
+    ).run()
+
+    expect(await new AppSettings().loadSelectedCity()).toBeFalsy()
```

Comment:
Maybe you should set some other city and then check that it's not been changed. Usually, we can only peek, if we have selected some city before.